docs/DeveloperGuide.md

```diff
@@ -1,38 +1,442 @@
+---
+
+[//]: # (@@author ollayf)
+# Design & Implementation
+
+
+## Architecture
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/ArchitectureDiagram.png" width="100%" />
+
+
+This **Architecture Diagram** explains the high-level design of the App.
+
+
+### Main Components of the architecture
+
+**SecureNUS** has 1 class called SecureNUS. It is responsible for,
+* At app launch: initializing all the important components in the correct sequence and initializes the appropriate connections between them.
+* It connects and sends the appropriate information and executes different actions using the other helpers via **Commands**.
+
+**SecretStorage** is responsible for the management of Secret Objects which store information of different Secret classes, e.g. passwords, credit cards. It also manages the different underlying data structures within it that help optimize the speed of queries through indexing, etc. It consists of 3 classes: **SecretMaster, SecretSearcher** and **SecretEnumerator**, which contain the different views.
+
+
+
+* **SecretMaster:** Manages the other views and also ensures there are no duplicates in naming where they should not exist
+* **SecretSearcher:** Contains the Hashtable view for directly querying the Secret by name
+* **SecretEnumerator:** Contains the ArrayList view for querying the Secret by index
+
+The other 2 components are:
+
+
+
+* **UI:** The main interface the user have access to through the Command Line
+* **Backend:** Which handles the details regarding the initialisation and saving (or exporting) of data across sessions
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/ArchitectureDiagram.png" width="100%" />  
+
+___
+### How the architecture components interact with each other
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user creates a new basic password initiated using the command `new password.`
+
+Sequence Diagram
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/SequenceDiagramNew.png" width="100%" />
+
+Each of the six main components (also shown in the diagram above)
+
+
+
+* defines its API in an interface with the same name as the Component.
+
+
+## UI Component
+
+The API of this component is specified in `Ui.java`
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/LightBulb.png" width="10%" />
+
+
+The UI consists of the Ui class which handles all the inputs provided by the user. The Ui then sends information to the 
+parser which will interpret the commands provided by user input. The **Parser **will create **Commands** which will be
+sent to **SecureNUS** to execute some command that will alter the data in SecretStorage. The **Ui **also handles 
+responses/ feedback to the User.
+
+
+## SecureNUS Component
+
+This is the main component that initialises all other components and connects them when the application is running.
+
+## SecretStorage Component
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/SecretStorageComponent.png" width="100%" />
```

Comment:
Perhaps it would be easier to read if the words 'requests alteration of data via'  and 'sends data for export OR collects data for import' were inside the box instead? (Such that there won't be lines cutting through the words)